index.js

```diff
@@ -15,7 +15,7 @@ var CR = 13;
  * @return {Object} The generator object
  */
 function* readlines(fd, filesize, bufferSize, position) {
-  if (typeof bufferSize === 'undefined') bufferSize = 1024;
+  if (typeof bufferSize === 'undefined') bufferSize = 64 * 1024;
```

Comment:
The reason being that this is much faster with larger files as we do not hit the disk that often. In addition, default page size is 4KiB when considering memory mapped disk I/O.

For smaller files, the guard you installed that reduces the buffer size to the remaining size will limit the read chunk buffer size to the actual file size.

Furthermore, for fs.createReadStream() the default buffer size is also 64KiB.

So, for files smaller than 64KiB, the whole file content will be read in one single read operation, increasing performance by approx. 4-5 ms without any of the other optimizations in place.